libnetwork/portmapper/proxy_linux.go

```diff
@@ -61,6 +64,15 @@ func (p *proxyCommand) Start() error {
 	}
 	w.Close()
 
+	go func() {
+		p.mu.Lock()
+		if p.exitedWait == false {
```

Comment:
Not sure this is correct, as this would keep the lock until the process exits, correct?

Perhaps (but I would have to check), the `p.cmd.Wait()` can be removed from `Stop()`, so that there's only a single Wait (new code does mean we'd always keep a go func running for the duration of the proxy; perhaps not an issue)

Reply:
Ah, but then we'd miss the error again.  Hm..

Reply:
can I delete p.cmd.Wait()  from Stop(), and add  some log  ?

Reply:
but stop() may run before 
go func() {
			err = p.cmd.Wait()
                }

Reply:
ping @tianon